Description: Handle EXTRA_* flags into icedtea
 - Use -J-Xbootclasspath for javac of PulseAudio
 - Append $EXTRA_CFLAGS_IT to $IT_CFLAGS (use for PulseAudio native build)
 - Append $EXTRA_LDFLAGS_IT to PulseAudio native link
 - Send all EXTRA_* flags to jamvm and cacao
 - Remove `-icedtea' from the release identifier
Author: Damien Raude-Morvan <drazzib@debian.org>, Matthias Klose <doko@ubuntu.com>
 
--- Makefile.am.orig
+++ Makefile.am
@@ -118,7 +118,7 @@
 IT_JAVACFLAGS=$(IT_JAVAC_SETTINGS) -source $(IT_LANGUAGE_SOURCE_VERSION) -target $(IT_CLASS_TARGET_VERSION)
 
 # Flags
-IT_CFLAGS=$(CFLAGS) $(ARCHFLAG)
+IT_CFLAGS=$(CFLAGS) $(ARCHFLAG) $(EXTRA_CFLAGS_IT)
 
 # Conditional defintions
 
@@ -1393,9 +1393,6 @@
 	  echo "DISTRO_PACKAGE_VERSION=$(PKGVERSION)" \
 	    >>openjdk/jdk/make/common/shared/Defs.gmk ;
 endif
-	if test x"$(PROJECT_NAME)" != "xjdk7"; then \
-	  proj_suffix="-$(PROJECT_NAME)"; \
-	fi ; \
 	if test x"$(VERSION_SUFFIX)" != "x"; then \
 	  ver_suffix="-$(VERSION_SUFFIX)"; \
 	fi ; \
@@ -1839,15 +1839,15 @@
 stamps/pulse-java-headers.stamp: stamps/pulse-java-class.stamp
 	mkdir -p $(PULSE_JAVA_NATIVE_BUILDDIR)
 	$(BOOT_DIR)/bin/javah -d $(PULSE_JAVA_NATIVE_BUILDDIR) -classpath $(PULSE_JAVA_CLASS_DIR) \
-	  org.classpath.icedtea.pulseaudio.EventLoop ;
+	  -J-Xbootclasspath/p:$(PULSE_JAVA_CLASS_DIR) org.classpath.icedtea.pulseaudio.EventLoop ;
 	$(BOOT_DIR)/bin/javah -d $(PULSE_JAVA_NATIVE_BUILDDIR) -classpath $(PULSE_JAVA_CLASS_DIR) \
-	  org.classpath.icedtea.pulseaudio.Stream
+	  -J-Xbootclasspath/p:$(PULSE_JAVA_CLASS_DIR) org.classpath.icedtea.pulseaudio.Stream
 	$(BOOT_DIR)/bin/javah -d $(PULSE_JAVA_NATIVE_BUILDDIR) -classpath $(PULSE_JAVA_CLASS_DIR) \
-	  org.classpath.icedtea.pulseaudio.Operation
+	  -J-Xbootclasspath/p:$(PULSE_JAVA_CLASS_DIR) org.classpath.icedtea.pulseaudio.Operation
 	$(BOOT_DIR)/bin/javah -d $(PULSE_JAVA_NATIVE_BUILDDIR) -classpath $(PULSE_JAVA_CLASS_DIR) \
-	  org.classpath.icedtea.pulseaudio.PulseAudioSourcePort
+	  -J-Xbootclasspath/p:$(PULSE_JAVA_CLASS_DIR) org.classpath.icedtea.pulseaudio.PulseAudioSourcePort
 	$(BOOT_DIR)/bin/javah -d $(PULSE_JAVA_NATIVE_BUILDDIR) -classpath $(PULSE_JAVA_CLASS_DIR) \
-	  org.classpath.icedtea.pulseaudio.PulseAudioTargetPort
+	  -J-Xbootclasspath/p:$(PULSE_JAVA_CLASS_DIR) org.classpath.icedtea.pulseaudio.PulseAudioTargetPort
 	mkdir -p stamps
 	touch $@
 
@@ -1856,7 +1856,7 @@
 	 -I$(PULSE_JAVA_NATIVE_BUILDDIR) -o $@ -c $<
 
 $(PULSE_JAVA_NATIVE_BUILDDIR)/libpulse-java.so: $(PULSE_JAVA_NATIVE_OBJECTS)
-	$(CC) $(LDFLAGS) -shared $(PULSE_JAVA_NATIVE_OBJECTS) $(LIBPULSE_LIBS) \
+	$(CC) $(LDFLAGS) $(EXTRA_LDFLAGS_IT) -shared $(PULSE_JAVA_NATIVE_OBJECTS) $(LIBPULSE_LIBS) \
 	 -o $(PULSE_JAVA_NATIVE_BUILDDIR)/libpulse-java.so
 
 endif
@@ -1924,7 +1924,8 @@
 if BUILD_JAMVM
 	cd jamvm/jamvm && \
 	./autogen.sh --with-java-runtime-library=openjdk \
-	  --prefix=$(abs_top_builddir)/jamvm/install ; \
+	  --prefix=$(abs_top_builddir)/jamvm/install \
+	  CFLAGS='$(EXTRA_CFLAGS_JAMVM)' LDFLAGS='$(EXTRA_LDFLAGS_JAMVM)' CPPFLAGS='$(EXTRA_CPPFLAGS_JAMVM)' CXXFLAGS='$(EXTRA_CXXFLAGS_JAMVM)'; \
 	$(MAKE) ; \
 	$(MAKE) install
 	mkdir -p $(abs_top_builddir)/jamvm/install/hotspot/jre/lib/$(INSTALL_ARCH_DIR)/server
@@ -1993,7 +1994,8 @@
 	  --with-java-runtime-library=openjdk7 \
 	  --with-java-runtime-library-prefix=$(abs_top_builddir)/openjdk \
 	  --with-java-runtime-library-classes=$(RUNTIME) \
-	  --enable-jre-layout $(CACAO_CONFIGURE_ARGS); \
+	  --enable-jre-layout $(CACAO_CONFIGURE_ARGS) \
+	  CFLAGS='$(EXTRA_CFLAGS_CACAO)' LDFLAGS='$(EXTRA_LDFLAGS_CACAO)' CPPFLAGS='$(EXTRA_CPPFLAGS_CACAO)' CXXFLAGS='$(EXTRA_CXXFLAGS_CACAO)'; \
 	$(ARCH_PREFIX) $(MAKE) -j$(PARALLEL_JOBS) install
 	ln -sf server $(abs_top_builddir)/cacao/install/jre/lib/$(INSTALL_ARCH_DIR)/client
 	touch $(abs_top_builddir)/cacao/install/jre/lib/$(INSTALL_ARCH_DIR)/server/Xusage.txt
--- configure.ac.orig	2012-04-02 22:18:07.000000000 +0200
+++ configure.ac	2012-04-02 22:18:42.000000000 +0200
@@ -196,14 +196,14 @@
 AC_SUBST(XT_LIBS)
 
 dnl Check for libXp headers and libraries.
-PKG_CHECK_MODULES(XP, xp,[XP_FOUND=yes],[XP_FOUND=no])
-if test "x${XP_FOUND}" = xno
-then
-  AC_MSG_ERROR([Could not find Xp - \
-  Try installing libXp-devel.])
-fi
-AC_SUBST(XP_CFLAGS)
-AC_SUBST(XP_LIBS)
+dnl PKG_CHECK_MODULES(XP, xp,[XP_FOUND=yes],[XP_FOUND=no])
+dnl if test "x${XP_FOUND}" = xno
+dnl then
+dnl   AC_MSG_ERROR([Could not find Xp - \
+dnl  Try installing libXp-devel.])
+dnl fi
+dnl AC_SUBST(XP_CFLAGS)
+dnl AC_SUBST(XP_LIBS)
 
 dnl Check for libX11 headers and libraries.
 PKG_CHECK_MODULES(X11, x11,[X11_FOUND=yes],[X11_FOUND=no])
